### PR TITLE
Use UICulture

### PR DESCRIPTION
**Pull request by tfenster**
Use UICulture to get localized client captions

### CODE REVIEW

**Comment by wortho** [comment]
Looking at this.

File: Microsoft.Dynamics.Nav.UserSession/ClientSessionExtensions.cs [hunk 1]
```diff
@@ -29,9 +29,11 @@ public static void InvokeInteraction(this ClientSession clientSession, ClientInt
 
         /// <summary>Opens the session synchronously.</summary>
         /// <param name="clientSession">The client Session.</param>
-        public static void OpenSession(this ClientSession clientSession)
+        public static void OpenSession(this ClientSession clientSession, string uiCulture = null)
         {
-            var sessionParameters = new ClientSessionParameters {CultureId = CultureInfo.CurrentCulture.Name};
+            var sessionParameters = new ClientSessionParameters { CultureId = CultureInfo.CurrentCulture.Name };
+            if (uiCulture != null)
+                sessionParameters.UICultureId = CultureInfo.GetCultureInfoByIetfLanguageTag(uiCulture).Name;
```
Comment (wortho) [inline Microsoft.Dynamics.Nav.UserSession/ClientSessionExtensions.cs]: This method is deprecated.   See https://msdn.microsoft.com/query/dev14.query?appId=Dev14IDEF1&l=EN-US&k=k(System.Globalization.CultureInfo.GetCultureInfoByIetfLanguageTag);k(TargetFrameworkMoniker-.NETFramework,Version%3Dv4.5);k(DevLang-csharp)&rd=true

Actually theres no need to do this as the parameter is the Culture Name.

File: Microsoft.Dynamics.Nav.LoadTest/app.config [hunk 2]
```diff
@@ -18,4 +21,11 @@
             </setting>
         </Microsoft.Dynamics.Nav.LoadTest.Properties.Settings>
     </applicationSettings>
+    <userSettings>
+        <Microsoft.Dynamics.Nav.LoadTest.Properties.Settings>
+            <setting name="UICulture" serializeAs="String">
+                <value>de-DE</value>
```
Comment (wortho) [inline Microsoft.Dynamics.Nav.LoadTest/app.config]: We can't add this to the main repro as it won't work if you don't have de-DE.

File: Microsoft.Dynamics.Nav.LoadTest/OrderProcessorScenarios.cs [hunk 3]
```diff
@@ -36,7 +36,8 @@ private UserContextManager CreateUserContextManager()
                     NAVClientService,
                     null,
                     null,
-                    OrderProcessorRoleCenterId);
+                    OrderProcessorRoleCenterId,
+                    UICulture);
```
Comment (wortho) [inline Microsoft.Dynamics.Nav.LoadTest/OrderProcessorScenarios.cs]: I made it an optional parameter so we don't need to add it to the example scenarios.

**Comment by wortho** [comment]
HI Tobias. I made some comments and submitted the core of your changes to the main branch for NAV2016. If you want to make some similar changes then I can merge your branch with the NAV2013R2 repo.

**Comment by tfenster** [comment]
Sorry, I am still not 100% sure how git works. This means that I'll change my adjustments in the same way as you did and the create a pull request again?

**Comment by wortho** [comment]
Hi @tfenster, I agree that it takes time to learn Git, but it is worth it in the end.
It is also a bit confusing with the different NAV versions as well. 
There should be a release branch for each NAV version. I don't have the time to maintain a branch for all 3 versions but if you want we could create an up-to-date release/nav2013R2 where you can merge any changes you need from the main branch.

**Comment by tfenster** [comment]
Yes, that would be nice. I guess I would then base my specific branch with the test classes for our solution on the release/nav2013r2 branch?

**Pull request closed without merging**